catalog/spatial/geowebcache/geowebcache-admin-plugin/src/main/webapp/js/model/GwcLayers.js

```diff
@@ -0,0 +1,163 @@
+                                that.trigger('change:layers', this);
+                            }
+                        });
+                    }
+                });
+            },
+            postLayerData: function(layerData, type) {
+                var restUrl = "../../geowebcache/rest/layers/" + layerData.name + ".xml";
+                var that = this;
+                var data = this.generateXmlForLayer(layerData);
+                $.ajax({
+                    url: restUrl,
+                    type : type,
+                    dataType : "xml",
+                    data : data,
+                    success: function() {
+                        that.getLayers();
+                    }
+                });
+            },
+            addLayer: function(layerData) {
+                this.postLayerData(layerData, "PUT");
+            },
+            updateLayer: function(layerData) {
+                if(this.nameExistsInArray(layerData.name)) {
+                    this.postLayerData(layerData, "POST");
+                } else {
+                    this.deleteLayer(layerData.oldName);
+                    this.postLayerData(layerData, "PUT");
+                }
+            },
+            nameExistsInArray: function(name) {
+                var exists = false;
+                $.each(this.get("layers"), function(index, value) {
+                    if(name === value.name) {
+                        exists = true;
+                    }
+                });
+                return exists;
+            },
+            getLayer: function(name) {
+                var layer;
+                $.each(this.get("layers"), function(index, value) {
+                    if(name === value.name) {
+                        layer = value;
+                    }
+                });
+                return layer;
+            },
+            generateXmlForLayer: function(layerData) {
+                var data;
+                data = "<wmsLayer>";
+                data+= "    <name>" + layerData.name + "</name>";
+                data+= "    <mimeFormats>";
+                $.each(layerData.mimes, function(index,value) {
+                    data +=  "      <string>" + value + "</string>";
+                });
+                data+= "    </mimeFormats>";
+                data+= "    <wmsUrl>";
+                data+= "        <string>" + layerData.url + "</string>";
+                data+= "    </wmsUrl>";
+                if(layerData.wmsLayerNames.length > 0) {
+                    data+="    <wmsLayers>";
+                    $.each(layerData.wmsLayerNames, function(index,value) {
+                        data +=  value;
+                         if(index < layerData.wmsLayerNames.length -1) {
+                            data+=",";
+                         }
+                    });
+                    data+="    </wmsLayers>";
+                }
+                data+= "</wmsLayer>";
```

Comment:
My above comment may be rendered null and void based on how this XML is being built up using the loops. If it becomes too much of a pain then not a big deal to leave it be.